Improves instructions for coding challenge (#2)
zeller-public/code-challenge-android-mobile: IN_PERSON_INSTRUCTIONS.md

# In Person Code Review Interview

> Use the instructions below if you have been invited to an In Person Code Review Interview. 

Reviewer: ```suggestion
# In-Person Code Review Interview

Use the instructions below if you have been invited to an In-Person Code Review Interview.
```

## Scenario

> This project represents a simple banking application. It allows a user to see a balance, deposit funds, and withdraw funds.

Reviewer: ```suggestion
## Scenario
This project contains a simple banking application. 
It allows a user to view their balance, deposit funds, and withdraw funds.
```

## Your Task
Please take some time to review the provided code before our interview. There is no need to write or refactor any code beforehand. Simply come prepared to discuss it.

> During the interview, we will talk through the code together and explore how you might evolve it into a more robust, scalable, and testable feature.

Reviewer: ```suggestion
## Your Task
Please take some time to review the provided code before your interview. If possible, run the app yourself and explore its functionality.
There is no need to write or refactor any code beforehand. Simply come prepared to discuss it.

During the interview, we will talk through the code together and explore how you might evolve it to support robust, scalable, and testable features.
```

## Expectations

> This is not a pass/fail coding test. The goal is not to find bugs or to have you write a perfect solution on the spot. Instead, we want to use this code as a starting point for a conversation about modern Android development practices, architectural principles, and trade-offs.

Reviewer: ```suggestion
## Expectations
This is not a pass/fail coding test. The goal is not to exhaustively criticise this sample, or to have you deliver a perfect re-write on the spot. 
Instead, we want to use this code as a starting point for a conversation about modern Android development practices, architectural principles, and trade-offs.
```

> ## What we're looking for

Reviewer: ```suggestion
## What We're Looking for
```

## Scenario
This project contains a simple banking application. 
It allows a user to view their balance, deposit funds, and withdraw funds.

## Your Task
Please take some time to review the provided code before your interview. If possible, run the app yourself and explore its functionality.
There is no need to write or refactor any code beforehand. Simply come prepared to discuss it.

During the interview, we will talk through the code together and explore how you might evolve it to support robust, scalable, and testable features.

## Expectations
This is not a pass/fail coding test. The goal is not to find bugs or to have you write a perfect solution on the spot. Instead, we want to use this code as a starting point for a conversation about modern Android development practices, architectural principles, and trade-offs.

## What we're looking for
As you review the code, think about it from a high level. Consider the following aspects:

- Separation of Concerns
- State Management
- Lifecycle Awareness
- Testability
- Robustness & Error Handling

## What NOT to Focus On
To help you save time and focus your efforts, please do not worry about the following:

> - Building a complete UI or fixing XML layouts.

Reviewer: ```suggestion
## What NOT to Focus on
To help you save time and focus your efforts, please do not worry about the following:

- Building a complete UI or fixing XML layouts.
- Writing a fully refactored version of the code before the interview.
- Using the absolute latest libraries from Google (e.g., Jetpack Compose, Hilt). 
    While you're welcome to discuss them as part of a potential solution, the conversation will be focused on the underlying architectural principles, not specific library choices.
```

Instead, we want to use this code as a starting point for a conversation about modern Android development practices, architectural principles, and trade-offs.

## What we're looking for
As you review the code, think about it from a high level. Consider the following aspects:

- Separation of Concerns
- State Management
- Lifecycle Awareness
- Testability
- Robustness & Error Handling

## What NOT to Focus On
To help you save time and focus your efforts, please do not worry about the following:

- Building a complete UI or fixing XML layouts.
- Writing a fully refactored version of the code before the interview.
- Using the absolute latest libraries from Google (e.g., Jetpack Compose, Hilt). While you're welcome to discuss them as part of a potential solution, the conversation will be focused on the underlying architectural principles, not specific library choices.

## Questions?
The discussion will be a collaborative, white-boarding style conversation with our Tech Lead. Your thought process and ability to articulate architectural trade-offs are what we're most interested in.

> 

Reviewer: ```suggestion
## Questions?
The discussion will be a collaborative, white-boarding style conversation with a Tech Lead. 
Your thought process and ability to articulate (trade-offs, best practice, prior experience, etc.) are what we're most interested in.

If you have any questions, please don't hesitate to send us an e-mail at tim@myzeller.com

Good luck and happy coding!

```

# In-Person Code Review Interview

Use the instructions below if you have been invited to an In-Person Code Review Interview.

## Scenario
This project contains a simple banking application. 
It allows a user to view their balance, deposit funds, and withdraw funds.

## Your Task
Please take some time to review the provided code before your interview. If possible, run the app yourself and explore its functionality.
There is no need to write or refactor any code beforehand. Simply come prepared to discuss it.

During the interview, we will talk through the code together and explore how you might evolve it to support robust, scalable, and testable features.

## Expectations
This is not a pass/fail coding test. The goal is not to exhaustively criticise this sample, or to have you deliver a perfect re-write on the spot. 
Instead, we want to use this code as a starting point for a conversation about modern Android development practices, architectural principles, and trade-offs.

## What We're Looking for
As you review the code, think about it from a high level. Consider the following aspects:

- Separation of Concerns
- State Management
- Lifecycle Awareness
- Testability

> ## What NOT to Focus on

Reviewer: ```suggestion
- Testability
- Robustness & Error Handling

## What NOT to Focus on
```

To help you save time and focus your efforts, please do not worry about the following:

- Building a complete UI or fixing XML layouts.
- Writing a fully refactored version of the code before the interview.
- Using the absolute latest libraries from Google (e.g., Jetpack Compose, Hilt). 
    While you're welcome to discuss them as part of a potential solution, the conversation will be focused on the underlying architectural principles, not specific library choices.

> ## Questions?

Reviewer: ```suggestion
    While you're welcome to discuss them as part of a potential solution, the conversation will be focused on the underlying architectural principles, not specific library choices.

## Questions?
```
